x-pack/plugins/apm/server/routes/alerts/rule_types/anomaly/register_anomaly_rule_type.ts

```diff
@@ -102,6 +106,14 @@ export function registerAnomalyRuleType({
         if (!ml) {
           return {};
         }
+
+        const config = await firstValueFrom(config$);
+        const indices = await getApmIndices({
+          config,
+          savedObjectsClient: services.savedObjectsClient,
+        });
+        const { transaction: transactionsIndex } = indices;
+
```

Comment:
I suggest you just pass the config observable to `getAnomalousEventSourceFields` and then do this from there.